tests/UnusedCodeTest.php

```diff
@@ -99,6 +104,70 @@ public function testInvalidCode($code, $error_message, $error_levels = []): void
         \Psalm\IssueBuffer::processUnusedSuppressions($this->project_analyzer->getCodebase()->file_provider);
     }
 
+    public function testSeesClassesUsedAfterUnevaluatedCodeIssue(): void
+    {
+        $this->project_analyzer->getConfig()->throw_exception = false;
+        $file_path = getcwd() . DIRECTORY_SEPARATOR . 'tests' . DIRECTORY_SEPARATOR . 'somefile.php';
```

Comment:
btw: the other tests in this file use a broken `$file_path` and only work by accident because the project analyzer is configured to throw rather than record the issue.

the other tests use `self::$src_dir_path . 'somefile.php';` which points to the `src ` dir which isn't in the project directory used by the test `project_analyzer` (which is `getcwd() . '/tests'` which causes it to not record any issues.

This took me a while to find, especially because everything is nicely failing silently along the path once you tell the config to not throw exceptions.